x-pack/plugin/async/src/main/java/org/elasticsearch/xpack/async/AsyncResultsIndexPlugin.java

```diff
@@ -82,7 +84,8 @@ public String getFeatureDescription() {
                 client,
                 ASYNC_SEARCH_ORIGIN,
                 AsyncSearchResponse::new,
-                namedWriteableRegistry
+                namedWriteableRegistry,
+                bigArrays
```

Comment:
I think that `AsyncTaskMaintenanceService` only uses this service to grab a client. I wonder if we could construct the client directly, to avoid creating this whole service? That would avoid updating the `Plugin#createComponents` signature.

Reply:
Good find. I will fix it.

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/73862/commits/78b7ab9a8a1bf91fa08ac6e3bfa00cb578672474.